### PR TITLE
Using the primitive variable index classes in more places (+ minor chores)

### PR DESCRIPTION
**Pull request by pcarruscag**
Cleanup related with #1392

### CODE REVIEW

File: Common/src/geometry/CPhysicalGeometry.cpp [hunk 1]
```diff
@@ -4752,78 +4752,80 @@ void CPhysicalGeometry::SetPoint_Connectivity() {
...
-    auto Degree = nodes->GetnPoint(iPoint);
-    if (Degree < MinDegree) { MinDegree = Degree; AddPoint = iPoint; }
+  /*--- Exclude halo nodes from the ordering process. ---*/
+  for (auto iPoint = nPointDomain; iPoint < nPoint; iPoint++) {
+    InQueue[iPoint] = true;
   }
 
-  /*--- Add the node in the first free position. ---*/
+  /*--- Repeat as many times as necessary to handle disconnected graphs. ---*/
+  while (Result.size() < nPointDomain) {
```
Comment (pcarruscag) [inline Common/src/geometry/CPhysicalGeometry.cpp:4771]: Reason for the residual changes.

File: SU2_CFD/include/solvers/CFVMFlowSolverBase.inl [hunk 2]
```diff
@@ -2559,15 +2550,9 @@ void CFVMFlowSolverBase<V, FlowRegime>::Friction_Forces(const CGeometry* geometr
...
-          else       Grad_Vel[iDim][jDim] = nodes->GetGradient_Primitive(iPoint, iDim + VEL_INDEX, jDim);
+          Grad_Vel[iDim][jDim] = nodes->GetGradient_Primitive(iPoint, prim_idx.Velocity() + iDim, jDim);
         }
-
-        /// TODO: Move the temperature index logic to a function.
-
-        if (FlowRegime == ENUM_REGIME::COMPRESSIBLE) Grad_Temp[iDim] = nodes->GetGradient_Primitive(iPoint, 0, iDim);
-
-        if (FlowRegime == ENUM_REGIME::INCOMPRESSIBLE) Grad_Temp[iDim] = nodes->GetGradient_Primitive(iPoint, nDim + 1, iDim);
+        Grad_Temp[iDim] = nodes->GetGradient_Primitive(iPoint, prim_idx.Temperature(), iDim);
```
Comment (WallyMaier) [inline SU2_CFD/include/solvers/CFVMFlowSolverBase.inl:2555]: Nice! 👍

**Review by WallyMaier — COMMENTED** [review]
LGTM! Thanks!

File: Common/src/geometry/CPhysicalGeometry.cpp (hunk 1 above)
Comment (TobiKattmann) [reply Common/src/geometry/CPhysicalGeometry.cpp:4771]: -> (from dev meeting) Halo points should be at the end of the point list as a convention.
The disconnected domain parts were the change (as they happen in the CHT pin case where disconnected pin-pieces are treated as 1 zone)

File: TestCases/species_transport/passive_transport_validation/passive_transport.cfg [hunk 3]
```diff
@@ -82,7 +82,6 @@ TIME_DISCRE_FLOW= EULER_IMPLICIT
 %
 % --------------------------- CONVERGENCE PARAMETERS --------------------------%
 %
-CONV_CRITERIA= RESIDUAL
```
Comment (TobiKattmann) [inline TestCases/species_transport/passive_transport_validation/passive_transport.cfg:85]: Wasn't me - Shaggy, 2000

**Review by TobiKattmann — COMMENTED** [review]
:bouquet: many good cleanups where hardcoded indexes were used. And good that CONV_CRITERIA is now deprecated

File: SU2_CFD/include/variables/CIncEulerVariable.hpp [hunk 4]
```diff
@@ -55,6 +55,9 @@ class CIncEulerVariable : public CFlowVariable {
     inline IndexType ThermalConductivity() const { return nDim+6; }
     inline IndexType CpTotal() const { return nDim+7; }
     inline IndexType CvTotal() const { return nDim+8; }
+
+    /*--- For compatible interface with NEMO. ---*/
+    inline IndexType Temperature_ve() const { return std::numeric_limits<IndexType>::max(); }
```
Comment (TobiKattmann) [inline SU2_CFD/include/variables/CIncEulerVariable.hpp:60]: I need to 
```c++
#include <limits>
```
for this to work using gcc/11.1.0 + openmpi/4.1.2. as well as 11.2.0 + 4.0.5. The test-builds pass(ed) so it cannot be a major issue. 
[This stackoverflow post](https://stackoverflow.com/questions/4798936/numeric-limits-was-not-declared-in-this-scope-no-matching-function-for-call-t) gives resource that the limits header is used much less widely in gcc11 standard library headers -> requiring to explicitly include it.

And I guess we want to assure foward-compatibility ;)

**Pull request merged**